### ISSUE TITLE
Paru crashes with PkgNotFound when trying to update the system

### ISSUE THREAD

**Issue by kris7t** [Morganamilo/paru#356]
### Affected Version
**paru -V**
```
paru v1.5.1.r30.g8ff97c2 - libalpm v12.0.2
```

### Description

I have some installed packages that are neither in any repo nor the AUR (`PKGBUILD`s written for private use), and I'm not using `LocalRepo`. When trying to update packages, `paru` crashes while trying to unwrap a `PkgNotFound` error.

This seems similar to #347. However, this one triggers without any `LocalRepo`.

### Configuration

```ini
#
# $PARU_CONF
# /etc/paru.conf
# ~/.config/paru/paru.conf
#
# See the paru.conf(5) manpage for options

#
# GENERAL OPTIONS
#
[options]
PgpFetch
Devel
Provides
DevelSuffixes = -git -cvs -svn -bzr -darcs -always
BottomUp
RemoveMake = no
SudoLoop
UseAsk
CombinedUpgrade
#CleanAfter
UpgradeMenu
NewsOnUpgrade
SortBy = popularity
SearchBy = name-desc
#LocalRepo
#Chroot
#MovePkgs

#
# Binary OPTIONS
#
[bin]
FileManager = vifm
#MFlags = --skippgpcheck
```

### Output

```sh
$ paru
no new news
:: Synchronizing package databases...
 testing is up to date
 core is up to date
 extra is up to date
 community-testing is up to date
 community is up to date
 multilib-testing is up to date
 multilib is up to date
:: Looking for AUR upgrades
:: Looking for devel upgrades
6 testing/libxcrypt                 4.4.19-1        -> 4.4.20-1
5 community-testing/oath-toolkit    2.6.6-2         -> 2.6.7-1
4 community-testing/ruby-rdoc       6.3.0-3         -> 6.3.1-1
3 community/hugo                    0.82.0-1        -> 0.83.0-1
2 multilib-testing/lib32-libxcrypt  4.4.19-1        -> 4.4.20-1
1 aur/ungoogled-chromium            89.0.4389.128-1 -> 90.0.4430.85-1
:: Packages to exclude (eg: 1 2 3, 1-3):
:: 
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: PkgNotFound', src/upgrade.rs:337:42
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by Morganamilo** [comment]
Does it work on stable paru? I wonder if the fix for #347 broke this.

**Comment by Morganamilo** [comment]
It does look like the fix for #347 shadowed a variable and broke the stuff below. Should work now.
